File: output/llama_4_maverick/avatar/direct_translation/java/python/codeforces_7_B.py

import sys
t, m = map(int, sys.stdin.readline().split())
memory = [0] * m
allocIdx = 0
for _ in range(t):
    op = sys.stdin.readline().split()
    if op[0] == 'alloc':
        n = int(op[1])
        length = 0
        canAlloc = False
        for i in range(m):
            if memory[i] == 0:
                length += 1
            else:
                length = 0
            if length == n:
                canAlloc = True
                length = i - n + 1
                break
        if canAlloc:
            allocIdx += 1
            for i in range(length, length + n):
                memory[i] = allocIdx
            print(allocIdx)
        else:
            print('NULL')
    elif op[0] == 'erase':
        x = int(op[1])
        if x <= 0:
            print('ILLEGAL_ERASE_ARGUMENT')
            continue
        hasErased = False
        for i in range(m):
            if memory[i] == x:
                memory[i] = 0
                hasErased = True
        if not hasErased:
            print('ILLEGAL_ERASE_ARGUMENT')
    elif op[0] == 'defragment':
        d = 0
        for i in range(m):
            if memory[i] == 0:
                d += 1
            else:
                memory[i - d] = memory[i]
        for i in range(m - d, m):
            memory[i] = 0
    else:
        print('h')

File: output/llama_4_maverick/avatar/direct_translation/python/java/codeforces_7_B.java

import java.util.Scanner;

public class codeforces_7_B {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        String[] line = scanner.nextLine().split(" ");
        int t = Integer.parseInt(line[0]);
        int m = Integer.parseInt(line[1]);
        int[] alocuente = new int[m];
        int ind = 1;

        for (int i = 0; i < t; i++) {
            line = scanner.nextLine().split(" ");
            if (line[0].equals("alloc")) {
                int pos = 0;
                int size = Integer.parseInt(line[1]);
                for (int j = 0; j < m; j++) {
                    if (alocuente[j] == 0) {
                        pos++;
                        if (pos == size) {
                            for (int k = j - size + 1; k <= j; k++) {
                                alocuente[k] = ind;
                            }
                            System.out.println(ind);
                            ind++;
                            break;
                        }
                    } else {
                        pos = 0;
                    }
                }
                if (pos != size) {
                    System.out.println("NULL");
                }
            }
            if (line[0].equals("erase")) {
                int arg = Integer.parseInt(line[1]);
                boolean found = false;
                for (int j = 0; j < m; j++) {
                    if (alocuente[j] == arg) {
                        found = true;
                        break;
                    }
                }
                if (!found || arg == 0) {
                    System.out.println("ILLEGAL_ERASE_ARGUMENT");
                } else {
                    for (int j = 0; j < m; j++) {
                        if (alocuente[j] == arg) {
                            alocuente[j] = 0;
                        }
                    }
                }
            }
            if (line[0].equals("defragment")) {
                int cnt = 0;
                for (int j = 0; j < m; j++) {
                    if (alocuente[j] == 0) {
                        cnt++;
                    }
                }
                int[] newAlocuente = new int[m];
                int idx = 0;
                for (int j = 0; j < m; j++) {
                    if (alocuente[j] != 0) {
                        newAlocuente[idx++] = alocuente[j];
                    }
                }
                for (int j = idx; j < m; j++) {
                    newAlocuente[j] = 0;
                }
                alocuente = newAlocuente;
            }
        }
    }
}

